crates/oxc_minifier/src/printer/gen.rs

```diff
@@ -102,7 +102,25 @@ fn print_if(if_stmt: &IfStatement<'_>, p: &mut Printer) {
     p.print(b'(');
     if_stmt.test.gen(p);
     p.print(b')');
-    if_stmt.consequent.gen(p);
+
+    match &if_stmt.consequent {
+        Some(Statement::BlockStatement(block)) => {
+            p.print_block1(block);
+        }
+        Some(stmt) if wrap_to_avoid_ambiguous_else(stmt) => {
+            p.print(b'{');
+            stmt.gen(p);
+            p.needs_semicolon = false;
```

Comment:
Can you add a comment as to why a semicolon is not needed here? It's hard to keep track.

Reply:
This is for avoiding `;` if the previous statement ends with `}`. I added the comment in `Printer::needs_semicolon` instead if inline here, because this is used in a lot of places.